src/module/item/sheet/rule-elements/token-light-form.ts

```diff
@@ -0,0 +1,60 @@
+import { LightData } from "types/foundry/common/data/data.js";
+import { RuleElementForm, RuleElementFormSheetData } from "./base.ts";
+import { TokenLightRuleElement, TokenLightRuleSource } from "@module/rules/rule-element/token-light.ts";
+import { isObject } from "remeda";
+
+class TokenLightForm extends RuleElementForm<TokenLightRuleSource, TokenLightRuleElement> {
+    override template = "systems/pf2e/templates/items/rules/token-light.hbs";
+
+    protected override tabNames = ["basic", "animation", "advanced"];
+    protected override tabDisplayStyle = "grid";
+
+    override async getData(): Promise<TokenLightSheetData> {
```

Comment:
What happens if all of this is moved to getInitialValue(), and it gets converted from private to protected?

Reply:
getInitialValue is called in every getData so I don't know if that is a good place. Maybe in the constructor?

Reply:
getData() is also called in every getData() though

Reply:
Maybe I misunderstood what you meant. I thought you were talking about the tab declarations.

Reply:
Oh, sorry, I mean the default settings stuff

Reply:
I'll look into it tomorrow.